src/ir/comp.rs

```diff
@@ -33,36 +33,44 @@ impl InterfaceSrc {
 #[derive(Default, Ctx)]
 /// A IR component. If `is_ext` is true then this is an external component.
 pub struct Component {
+    #[ctx(Expr)]
     // Interned data. We store this on a per-component basis because events with the
     // same identifiers in different components are not equal.
     /// Interned expressions
     exprs: Interned<Expr>,
     #[ctx(Time)]
+    #[add_ctx(Time)]
     /// Interned times
     times: Interned<Time>,
+    #[ctx(Prop)]
     /// Interned propositions
     props: Interned<Prop>,
 
     // Component defined values.
     #[ctx(Port)]
+    #[add_ctx(Port)]
     #[mut_ctx(Port)]
     /// Ports and bundles defined by the component.
     ports: IndexStore<Port>,
     #[ctx(Param)]
+    #[add_ctx(Param)]
     #[mut_ctx(Param)]
```

Comment:
It also might be worth combining this somehow, maybe with a macro for all ctx types or something?

Reply:
We can change the derive macro to instead support the syntax:
```
#[ctx(Param: add, get, mut)]
```
To signify that we want the `add`, `get`, and `mut` traits implemented.